src/posix/platform/openthread-posix-config.h

```diff
@@ -345,7 +345,7 @@
  *
  */
 #ifndef OPENTHREAD_POSIX_CONFIG_FACTORY_CONFIG_FILE
-#define OPENTHREAD_POSIX_CONFIG_FACTORY_CONFIG_FILE "src/posix/platform/openthread.conf.example"
+#define OPENTHREAD_POSIX_CONFIG_FACTORY_CONFIG_FILE "/tmp/openthread.conf.example"
```

Comment:
This here seem to specify the default path to use if `OPENTHREAD_POSIX_CONFIG_FACTORY_CONFIG_FILE` is not set.

Can we consider having `otbr` specify `OPENTHREAD_POSIX_CONFIG_FACTORY_CONFIG_FILE` with whatever path it wants (and has access to)? Will this work?

Reply:
The `ot-br-posix` uses the same config file `src/posix/platform/openthread-posix-config.h`, so we can't specify the  `OPENTHREAD_POSIX_CONFIG_FACTORY_CONFIG_FILE ` in the `ot-br-posix`'s config file. I also tried to specify the  `OPENTHREAD_POSIX_CONFIG_FACTORY_CONFIG_FILE` in the CMake file, the problem is that it is hard for developers to change it in `ot-br-posix` later. Thoughts?

Reply:
How about if we change it to behave similar to cmake `OT_CONFIG` and/or `OPENTHREAD_PROJECT_CORE_CONFIG_FILE` works for the core config.
With this next layer is expected to provide the config header file it wants (we can add a new `OT_POSIX_CONFIG` to cmake for posix?)

Then the `src/posix/platform/openthread-posix-config.h` becomes the default if the next module does not provide one?  (similar to how all the `src/config/{file}.h` are default and are not directly changed by next layer?)

Reply:
Based on your suggestion, I have pushed two commits to resolve this issue. Could you help review these two commits?
https://github.com/openthread/openthread/pull/8612
https://github.com/openthread/ot-br-posix/pull/1692

Reply:
Both LGTM. Thanks.